src/path.c

```diff
@@ -555,6 +558,32 @@ spherepath_get_point(PG_FUNCTION_ARGS)
 	PG_RETURN_NULL();
 }
 
+Datum
+spherepath_get_array(PG_FUNCTION_ARGS)
+{
+    SPATH *path = PG_GETARG_SPATH(0);
+    Datum *datum_arr = (Datum *) palloc(sizeof(Datum) * path->npts);
+    ArrayType *res;
+	SPoint *p = (SPoint *) palloc(sizeof(SPoint) * path->npts);
+
+    for (size_t i = 0; i < path->npts; i++)
+    {
+        if (!spath_get_point(&p[i], path, i))
+        {
+            // Clean up and return NULL
+            for (size_t j = 0; j < i; j++)
```

Comment:
Invalid memory free. Remove for loop. Just use pfree(p). It should be done like in spherepoly_get_array.